lib/AST/DeclContext.cpp

```diff
@@ -989,15 +989,15 @@ void IterableDeclContext::loadAllMembers() const {
   if (!hasLazyMembers())
     return;
 
-  // Don't try to load all members re-entrant-ly.
-  FirstDeclAndLazyMembers.setInt(false);
```

Comment:
@CodaFi maybe I'm confused about what's going on here, but it seems like this logic is not correct. Shouldn't we mark the members as lazily complete _after_ we load them?

Reply:
OK, after investigating some test failures, I see why this is needed now. Otherwise we get into a re-entrant loop where we keep adding the same members.